compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1627,24 +2051,30 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         decl: &FnDecl,
         fn_node_id: Option<NodeId>,
         kind: FnDeclKind,
+        generics: &Generics,
+        parent_generics: Option<&Generics>,
         make_ret_async: Option<NodeId>,
     ) -> &'hir hir::FnDecl<'hir> {
         let c_variadic = decl.c_variadic();
 
         // Skip the `...` (`CVarArgs`) trailing arguments from the AST,
         // as they are not explicit in HIR/Ty function signatures.
         // (instead, the `c_variadic` flag is set to `true`)
-        let mut inputs = &decl.inputs[..];
+        let mut ast_inputs = &decl.inputs[..];
         if c_variadic {
-            inputs = &inputs[..inputs.len() - 1];
+            ast_inputs = &ast_inputs[..ast_inputs.len() - 1];
         }
-        let inputs = self.arena.alloc_from_iter(inputs.iter().map(|param| {
+        let mut impl_trait_inputs = vec![];
+        let inputs = self.arena.alloc_from_iter(ast_inputs.iter().map(|param| {
             if fn_node_id.is_some() {
-                self.lower_ty_direct(&param.ty, ImplTraitContext::Universal)
+                self.lower_ty_direct(
+                    &param.ty,
+                    &mut ImplTraitContext::Universal { apit_nodes: &mut impl_trait_inputs },
```

Comment:
Nit: could we create the `itctxt` outside of the iterator, for both branches?

Reply:
Done https://github.com/rust-lang/rust/pull/101345/commits/971641f5307c81f4eac7d48c252b783a23a83ba4